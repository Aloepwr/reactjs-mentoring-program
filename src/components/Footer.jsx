import React from 'react';
import "../stylesheets/Footer.less";


const Footer = () => (
  <footer className="app-footer">
    <span className="span-title">netflixroulette</span>
  </footer>
);

export default Footer;