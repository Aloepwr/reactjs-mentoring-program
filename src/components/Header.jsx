import { Link } from 'react-router-dom';
import { withRouter } from 'react-router';


let searchQuery = "";

const Header = withRouter( ({history}) => (
  <header>
    <div className="header">
      <span className="span-title">netflixroulette</span>
      <section className="header__search-field">
        <h2>Find your movie</h2>
        <input type="search" placeholder="Enter the movie name here" onChange={ (e) => searchQuery = e.target.value }/>
      </section>
      <div className="header__search-sort">
        <ul>
          <li>Search by</li>
          <li><a href="#">Title</a></li>
          <li><a href="#">Director</a></li>
        </ul>
        <button className="btn__search" onClick={ () => history.push(`/search/${encodeURI(searchQuery)}`) }>
          Search
        </button>
      </div>
    </div>
  </header>
));

export default Header;