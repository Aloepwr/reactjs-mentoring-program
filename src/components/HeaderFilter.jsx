import React from 'react';
import "../stylesheets/HeaderFilter.less";


const HeaderFilter = (props) => (
  <div className="search-filter">
    { props.children }
  </div>
);

export default HeaderFilter;