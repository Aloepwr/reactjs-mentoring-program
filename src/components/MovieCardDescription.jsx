import React from 'react';
import { withRouter } from 'react-router';

import "../stylesheets/MovieCardDescription.less";


const MovieCardDescription = withRouter( (props) => {
  const { poster, title, vote_average, release_date, runtime, overview, budget, seasons, last_air_date } = props.currentMovie;
  
  const rating = <p>{ vote_average }</p>;

  return (
    <div className="movie-description">
      <div className="movie-description__head">
        <span className="span-title">netflixroulette</span>
        <button className="btn__back" onClick={ () => props.history.goBack() }>Search</button>
      </div>
      
      <div className="movie-description__layout">
        <div className="movie-description__image">
          { poster ? <img src={`https://image.tmdb.org/t/p/w185_and_h278_bestv2${poster}`} /> : null }
        </div>
        <div className="movie-description__details">
          <div className="movie-description__details--title">
            <h2>{ title }</h2>
            { vote_average ? rating : null }
          </div>
          <div className="movie-description__info">
            { release_date ? <p>{release_date}</p> : null }
            { runtime ? <p>{runtime} min</p> : null }
          </div>
          <p className="movie-description__summary">{overview}</p>
          <div className="movie-description__bottom-info">
            { budget ? <p>Budget is {budget}$</p> : null }
            { seasons ? <p>Seasons: {seasons}</p> : null }
            { last_air_date ? <p>Last air date: {last_air_date}</p> : null }
          </div>
        </div>
      </div>
    </div>
  );
});

export default MovieCardDescription;
