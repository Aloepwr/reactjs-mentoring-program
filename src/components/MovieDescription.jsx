// let movieInfo = {
//   "unit": 87,
//   "show_id": 60031236,
//   "show_title": "Kill Bill: Vol. 1",
//   "release_year": "2003",
//   "rating": "3.8",
//   "category": "Action & Adventure",
//   "show_cast": "Uma Thurman, Lucy Liu, Vivica A. Fox, Daryl Hannah, David Carradine, Michael Madsen, Julie Dreyfus, Chiaki Kuriyama, Sonny Chiba, Gordon Liu",
//   "director": "Quentin Tarantino",
//   "summary": "An assassin is shot by her ruthless employer, Bill, and other members of their assassination circle. But she lives -- and plots her vengeance.",
//   "img": "http://netflixroulette.net/api/posters/60031236.jpg",
//   "mediatype": 0,
//   "runtime": "111 min"
// }


class MovieDescription extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      movieInfo: {
        "unit": 87,
        "show_id": 60031236,
        "show_title": "Kill Bill: Vol. 1",
        "release_year": "2003",
        "rating": "3.8",
        "category": "Action & Adventure",
        "show_cast": "Uma Thurman, Lucy Liu, Vivica A. Fox, Daryl Hannah, David Carradine, Michael Madsen, Julie Dreyfus, Chiaki Kuriyama, Sonny Chiba, Gordon Liu",
        "director": "Quentin Tarantino",
        "summary": "An assassin is shot by her ruthless employer, Bill, and other members of their assassination circle. But she lives -- and plots her vengeance.",
        "img": "http://netflixroulette.net/api/posters/60031236.jpg",
        "mediatype": 0,
        "runtime": "111 min"
      }
    };
  }

  render() {
    return(
      <div className="movie-description" id="movie-description--popup">
        <div className="movie-description__image">
          <img src={this.state.movieInfo.img} />
        </div>
        <div className="movie-description__details">
          <div className="movie-description__details--title">
            <h2>{this.state.movieInfo.show_title}</h2>
            <p>{this.state.movieInfo.rating}</p>
          </div>
          <p>Oscar-winning Movies</p>
          <div className="movie-description__info">
            <p>{this.state.movieInfo.release_year}</p>
            <p>{this.state.movieInfo.runtime}</p>
          </div>
          <p className="movie-description__summary">{this.state.movieInfo.summary}</p>
          <div className="movie-description__people">
            <p>Director: {this.state.movieInfo.director}</p>
            <p>Cast: {this.state.movieInfo.show_cast}</p>
          </div>
        </div>
      </div>
    )
  }
};

export default MovieDescription;