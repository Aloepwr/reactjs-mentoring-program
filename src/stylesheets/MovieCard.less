//
// main container / movie card
//

.movie-card__list {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;

  max-width: 1050px;
  margin: 0 auto;
}

.movie-card {
  margin: 40px 0 30px;

  &:last-child {
    padding-bottom: 50px;
  }
}

.movie-card__layout {
  padding: 0 15px;
}

.movie-card__image {
  img {
    width: 100%;
    height: auto;
    cursor: pointer;
  }
}

.movie-card__description {
  display: flex;
  justify-content: space-between;
  
  padding: 12px 0;
  
  font-family: Arial;
  color: black;
}

.movie-card__description--title {
  h4 {
    font-size: 18px;
    padding-bottom: 7px;
  }

  p {
    font-size: 14px;
    color: #453F3D;
  }
}

.movie-card__description--year {
  div {
    border: 1px black solid;
    border-radius: 2px;
    padding: 3px 7px;
    font-size: 10px;
  }
}