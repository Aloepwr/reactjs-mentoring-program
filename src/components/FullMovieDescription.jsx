import React from 'react';
import { Link } from 'react-router-dom';

import HeaderFilter from './HeaderFilter';
import MoviesList from './MoviesList';
import MovieCardDescription from './MovieCardDescription';
import Footer from './Footer';

import '../stylesheets/MovieCardDescription.less';
import '../stylesheets/Footer.less';
import '../stylesheets/styles.less';


export class FullMovieDescription extends React.Component {
  componentWillMount() {
    this.props.loadMovieInfo(this.props.match.params.filmId, this.props.match.params.type);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.match.params.filmId != this.props.match.params.filmId) {
      this.props.loadMovieInfo(nextProps.match.params.filmId, this.props.match.params.type);
    }
  }

  render() {
    let movieListBar = <div className="search-filter__movies-amount">
                         <div>Films by {this.props.currentMovie.director}</div>
                       </div>

    return (
      <div>
        <header>
        <MovieCardDescription currentMovie={ this.props.currentMovie } />
        </header>
        <main>
          <HeaderFilter>
            { this.props.currentMovie.director ? movieListBar : null }
          </HeaderFilter>
          <MoviesList movies={ this.props.relatedMovies } />
        </main>
        <Footer />
      </div>
    );
  }
}