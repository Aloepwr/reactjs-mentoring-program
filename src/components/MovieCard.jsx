import { Link } from 'react-router-dom';
import "../stylesheets/MovieCard";

export function MovieCard(props) {
  const { poster, title, release_date, id, type } = props;

  return (
    <div className="movie-card">
      <div className="movie-card__layout">
        <div className="movie-card__image">
          <Link to={ `/${type}/${id}` }>
            <img src={ poster } />
          </Link>
        </div>
        <div className="movie-card__description">
          <div className="movie-card__description--title">
            <h4>{ title }</h4>
          </div>
          <div className="movie-card__description--year">
            <div>{ release_date }</div>
          </div>            
        </div>
      </div>
    </div>
  )
};

// let movieInfo = {
//   img: "",
//   title: "Kill Bill: Vol. 1",
//   genre: "Action & Adventure",
//   year: "2003"
// };

// let moviesList = [movieInfo];

// function movieItem(props, index) {
//   return (
//     <li className="movie-card" key={index}>
//       <div className="movie-card__layout">
//         <div className="movie-card__image">
//           {/* <img src={props.img} /> */}
//         </div>
//         <div className="movie-card__description">
//           <div className="movie-card__description--title">
//             <h4>{props.title}</h4>
//             <p>{props.genre}</p>
//           </div>
//           <div className="movie-card__description--year">
//             <div>{props.year}</div>
//           </div>            
//         </div>
//       </div>
//     </li>
//   );
// }

// const MovieCard = (props) => {
//   if (props.match.params.searchQuery) {
//     if (moviesList.length >= 1) { 
//       return (
//         <ul className="movie-card__list" onClick={ () => props.history.push(`/film/${encodeURI(props.match.params.searchQuery)}`) }>
//           {moviesList.map(movieItem)}
//         </ul>
//       );
//     }
//   } else {
//     return ( <h1 className="no-content">No films found</h1> );
//   }
// };

// export default MovieCard;


