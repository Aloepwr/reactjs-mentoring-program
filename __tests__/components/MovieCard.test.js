import renderer from "react-test-renderer";
import Adapter from 'enzyme-adapter-react-16';
import { mount, configure } from "enzyme";
import { MemoryRouter } from 'react-router-dom';
import { MovieCard } from "../../src/components/MovieCard";

configure({ adapter: new Adapter() });

const title = "test title";
const release_date = "1906-08-24";
const poster = "https://image.tmdb.org/t/p/w185_and_h278_bestv2/wajNlrVDkryVGTzDz9vZE6XKZnC.jpg";
const id = 111;
const type = "movie";

describe("Movie card component", () => {
  test("renders valid layout when it receives all props", () => {
    const tree = renderer.create(
      <MemoryRouter>
        <MovieCard title={ title } release_date={ release_date } poster={ poster } type={ type } id={ id }/>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders valid layout without release_date", () => {
    const tree = renderer.create(
      <MemoryRouter>
        <MovieCard title={ title } poster={ poster } type={ type } id={ id }/>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("constructs link for router", () => {
    const wrapper = mount(
      <MemoryRouter>
        <MovieCard type={ type } id={ id }/>
      </MemoryRouter>
    );

    expect(wrapper.find('[href="/movie/111"]').length).toBe(1);
  });
});