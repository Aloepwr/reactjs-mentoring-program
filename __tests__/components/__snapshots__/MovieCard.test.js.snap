// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie card component renders valid layout when it receives all props 1`] = `
<div
  className="movie-card"
>
  <div
    className="movie-card__layout"
  >
    <div
      className="movie-card__image"
    >
      <a
        href="/movie/111"
        onClick={[Function]}
      >
        <img
          src="https://image.tmdb.org/t/p/w185_and_h278_bestv2/wajNlrVDkryVGTzDz9vZE6XKZnC.jpg"
        />
      </a>
    </div>
    <div
      className="movie-card__description"
    >
      <div
        className="movie-card__description--title"
      >
        <h4>
          test title
        </h4>
      </div>
      <div
        className="movie-card__description--year"
      >
        <div>
          1906-08-24
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Movie card component renders valid layout without release_date 1`] = `
<div
  className="movie-card"
>
  <div
    className="movie-card__layout"
  >
    <div
      className="movie-card__image"
    >
      <a
        href="/movie/111"
        onClick={[Function]}
      >
        <img
          src="https://image.tmdb.org/t/p/w185_and_h278_bestv2/wajNlrVDkryVGTzDz9vZE6XKZnC.jpg"
        />
      </a>
    </div>
    <div
      className="movie-card__description"
    >
      <div
        className="movie-card__description--title"
      >
        <h4>
          test title
        </h4>
      </div>
      <div
        className="movie-card__description--year"
      >
        <div />
      </div>
    </div>
  </div>
</div>
`;
